ProjectLighthouse/Controllers/ReviewController.cs

```diff
@@ -74,5 +81,177 @@ public async Task<IActionResult> DPadRate(int slotId, [FromQuery] int rating)
+                    {
+                        {
+                            "hint_start", pageStart + pageSize
+                        },
+                        {
+                            "hint", pageStart // not sure
+                        },
+                    });
+            return this.Ok(response);
+        }
+
+        [HttpGet("reviewsBy/{username}")]
+        public async Task<IActionResult> ReviewsBy(string username, [FromQuery] int pageStart = 1, [FromQuery] int pageSize = 10) 
+        {
+            User? user = await this.database.UserFromRequest(this.Request);
+            if (user == null) return this.StatusCode(403, "");
+
+            Token? token = await this.database.TokenFromRequest(this.Request);
+            if (token == null) return this.StatusCode(403, "");
+
+            GameVersion gameVersion = token.GameVersion;
+
+            IEnumerable<Review> reviews = this.database.Reviews.Where(r => r.Reviewer.Username == username && r.Slot.GameVersion <= gameVersion)
+                .Include(r => r.Reviewer)
+                .Include(r => r.Slot)
+                .AsEnumerable() // performance?
+                .OrderByDescending(r => r.Timestamp)
+                .Skip(pageStart - 1)
+                .Take(pageSize);
+                
+            string inner = Enumerable.Aggregate(reviews, string.Empty, (current, review) => {
+                RatedLevel? ratedLevel = this.database.RatedLevels.FirstOrDefault(r => r.SlotId == review.SlotId && r.UserId == user.UserId);
+                RatedReview? ratedReview = this.database.RatedReviews.FirstOrDefault(r => r.ReviewId == review.ReviewId && r.UserId == user.UserId);
+                return current + review.Serialize(ratedLevel, ratedReview);
+            });
+
+            string response = LbpSerializer.TaggedStringElement("reviews", inner, new Dictionary<string, object>
+                    {
+                        {
+                            "hint_start", pageStart
+                        },
+                        {
+                            "hint", reviews.Last().Timestamp // Seems to be the timestamp of oldest
+                        },
+                    });
+
+            return this.Ok(response);
+        }
+
+        [HttpPost("rateReview/user/{slotId:int}/{username}")]
+        public async Task<IActionResult> RateReview(int slotId, string username, [FromQuery] int rating = 0) {
+            User? user = await this.database.UserFromRequest(this.Request);
+            if (user == null) return this.StatusCode(403, "");
+
+            User? reviewer = await this.database.Users.FirstOrDefaultAsync(u => u.Username == username);
+            if (reviewer == null) return this.StatusCode(403, "");
+
+            Review? review = await this.database.Reviews.FirstOrDefaultAsync(r => r.SlotId == slotId && r.ReviewerId == reviewer.UserId);
+            if (review == null) return this.StatusCode(403, "");
```

Comment:
These should be returning 400/404

Reply:
I went with 400, makes sense to me that a bad request would be trying to rate something that doesn't exist. https://github.com/LBPUnion/project-lighthouse/pull/62/commits/ed1eac696a11a3bfb2b477b0bfec4c153fcafecf